metagraph/src/annotation/binary_matrix/row_sparse_disk/row_sparse_disk.hpp

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "annotation/binary_matrix/base/binary_matrix.hpp"
+#include "common/vectors/bit_vector_adaptive.hpp"
+
+namespace mtg {
+namespace annot {
+namespace binmat {
+
+
+class RowSparseDisk : public BinaryMatrix {
+  public:
+    RowSparseDisk(size_t disk_buffer_size = 10'000) {
+        int_vector_buffer_params.buff_size = disk_buffer_size;
+    }
+
+    uint64_t num_columns() const override { return num_columns_; }
+    uint64_t num_rows() const override { return num_rows_; }
+
+    bool get(Row row, Column column) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get(row, column);
+    }
+    SetBitPositions get_row(Row row) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_row(row);
+    }
+    std::vector<Row> get_column(Column column) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_column(num_rows(), column);
+    }
```

Comment:
Maybe also print a warning saying that this is an extremely inefficient operation and one should consider converting to a column-major format?

Reply:
Agree, but maybe the same should be added to `row_sparse` and possibly other row-major formats? I added warn for this annotation.